converter/dicom-cast/src/Microsoft.Health.DicomCast.Core/Features/Worker/FhirTransaction/Observation/ObservationParser.cs

```diff
@@ -0,0 +1,354 @@
+            // Return the final aggregated list of observations.
+            foreach (DicomContentItem childItem in report.Children())
+            {
+                ParseObservations(childItem.Dataset, patientRef, imagingStudyRef, observations);
+            }
+        }
+
+        /// <summary>
+        /// Creates an Observation "Dose Summary" profile based on the the spec outlined at:
+        /// https://confluence.hl7.org/display/IMIN/Radiation+Dose+Summary+for+Diagnostic+Procedures+on+FHIR#RadiationDoseSummaryforDiagnosticProceduresonFHIR-DoseSummaryObservationProfile
+        /// </summary>
+        /// <returns>A Dose Summary</returns>
+        private static Observation CreateDoseSummary(
+            DicomDataset dataset,
+            ResourceReference imagingStudyRef,
+            ResourceReference patientRef)
+        {
+            // Create the observation
+            var observation = new Observation
+            {
+                // Set the code.coding
+                Code = new CodeableConcept("http://loinc.org", "73569-6", "Radiation exposure and protection information"),
+                // Add Patient reference
+                Subject = patientRef
+            };
+            // Add ImagingStudy reference
+            observation.PartOf.Add(imagingStudyRef);
+
+            // Set identifiers
+            // Attempt to get the StudyInstanaceUID from the report; if it is not there fallback to the Tag value in the dataset
+            var report = new DicomStructuredReport(dataset);
+            var studyInstanceUid = report.Get<string>(
+                ObservationConstants.StudyInstanceUid,
+                "");
+            if (string.IsNullOrEmpty(studyInstanceUid))
+            {
+                if (!dataset.TryGetSingleValue(DicomTag.StudyInstanceUID, out studyInstanceUid))
+                {
+                    throw new MissingMemberException($"Unable to {nameof(DicomTag.StudyInstanceUID)} from dose summary observation dataset");
+                }
+            }
+
+            Identifier studyInstanceIdentifier = ImagingStudyIdentifierUtility.CreateIdentifier(studyInstanceUid);
+            observation.Identifier.Add(studyInstanceIdentifier);
+
+            // Try to get accession number from report first then tag; ignore if it is not present it is not a required identifier.
+            var accessionNumber = report.Get<string>(ObservationConstants.AccessionNumber, "");
+            if (string.IsNullOrEmpty(accessionNumber))
+            {
+                dataset.TryGetSingleValue(DicomTag.AccessionNumber, out accessionNumber);
+            }
+
+            if (!string.IsNullOrEmpty(accessionNumber))
+            {
+                var identifier = new Identifier
+                {
+                    Value = accessionNumber,
+                    Type = new CodeableConcept("http://terminology.hl7.org/CodeSystem/v2-0203", "ACSN")
```

Comment:
nit: `FhirTransactionConstants.AccessionNumberTypeSystem` and `FhirTransactionConstants.AccessionNumberTypeCode`